xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +78,31 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights to items
+     * This functon is used in letor to update the weights assigned by the ranker
+     * It updates weights of all the elements in items as well as updating the max_attained
+     */
+    template <typename Iterator>
+    void set_new_weights(Iterator first, Iterator last) {
+	int i = 0;
+	if (std::distance(first, last) != size())
+	{
+	    printf("Number of weights assigned don't match the number of items.Aborting");
+	    return;
+	}
+	for (Iterator it = first;it != last;++it)
+	{
+	    set_new_weights_helper(*it, i);
+	    ++i;
+	}
+    }
+
+    /**
+     * Sorts the MSet::Internal::items according to their weights thereby sorting MSet
```

Comment:
Documentation for functions in the public API shouldn't generally mention the wrapped internal class. Just say something like "Sort the items in this MSet by weight. Use after calling set_new_weights()."